src/app/record/object-record/object-record.component.ts

```diff
@@ -0,0 +1,84 @@
+import { Component, OnInit, Input } from '@angular/core';
+import { JsonPatch } from '../../shared/interfaces';
+
+@Component({
+  selector: 'object-record',
+  templateUrl: './object-record.component.html',
+  styleUrls: ['./object-record.component.scss']
+})
+export class ObjectRecordComponent implements OnInit {
+  @Input() record: object;
+  @Input() diff?: JsonPatch[];
+  @Input() depth: number;
+  @Input() expanded: boolean;
+
+  keys: Array<string> = [];
+  expandObject = false;
+  diffType: string;
+  private isArray = false;
+  isExpandedMap = new Map<string,boolean>();
+  diffKeys = new Map<string,boolean>();
+
+  ngOnInit() {
+    this.keys = Object.keys(this.record);
+    if (this.diff) {
+      this.diff.forEach(item => {
+        if (item.path.length > this.depth) {
+          this.diffKeys.set(item.path[this.depth],true);
+          this.expand(item.path[this.depth]);
+        } else if (item.path.length <= this.depth) {
```

Comment:
isn't all items has the same path since they are for this particular object. so this check could be outside.

I am not sure though, if each component has diff for itself and all of its children or only for itself.